src/validation/statement.rs

```diff
@@ -145,20 +164,40 @@ fn validate_reference_expression<T: AnnotationMap>(
     base: &Option<Box<AstNode>>,
 ) {
     match access {
-        ReferenceAccess::Member(m) | ReferenceAccess::Global(m) => {
-            visit_statement(validator, m.as_ref(), context);
-
-            if let Some(reference_name) = statement.get_flat_reference_name() {
-                validate_reference(
-                    validator,
-                    statement,
-                    base.as_deref(),
-                    reference_name,
-                    &m.get_location(),
-                    context,
+        ReferenceAccess::Global(m) => {
+            if m.get_initial_base().or(Some(m)).is_some_and(|it| it.is_super() || it.has_super_metadata()) {
+                // super cannot be accessed as a global
+                validator.push_diagnostic(
+                    Diagnostic::new("`SUPER` is not allowed in global-access position.")
+                        .with_location(m.get_location())
+                        .with_error_code("E119"),
                 );
+            };
+
+            validate_member_access(validator, context, statement, m, base);
+        }
+        ReferenceAccess::Member(m) => {
+            // FIXME: `SUPER` access behind the global namespace operator is not caught. i.e. `.SUPER^...`, since we do not have a base here
```

Comment:
Is this outdated? The global namespace operator is caught in the above case, no?